app/models/electrical.rb

```diff
@@ -0,0 +1,43 @@
+class Electrical < ActiveRecord::Base
+  has_paper_trail class_name: 'Version'
+
+  belongs_to :country, required:false
+  has_one :package, as: :detail, dependent: :destroy
+  before_save :set_tested_on, if: :test_status_changed?
+  before_save :downcase_brand, if: :brand_changed?
+  before_save :set_updated_by
+  after_create :create_on_stockit
+  after_update :update_on_stockit
+
+  private
+
+  def set_tested_on
+    self.tested_on = Date.today
+  end
+
+  def downcase_brand
+    self.brand.downcase!
```

Comment:
If the `self.brand` value is already downcased, then  `self.brand.downcase!` will return `nil` and `before_save` callback will fail

```suggestion
    self.brand = self.brand.downcase
```

Reply:
oh, sure, I'll make that change

Reply:
yes @steveyken this is covered in the spec.